main.cpp

```diff
@@ -46,33 +107,49 @@ struct BoulderProblem
     {
         int holdType = 2;
         double holdSize = 0.8; 
+        double holdHeight;
     };
 
-    void calculateDifficulty( int problemGrade, Hold hold );
+    double calculateDifficulty( double ropeLength );
 
     Hold crimp;
 };
 
+double BoulderProblem::calculateDifficulty( double ropeLength )
+{
+    double difficulty = ( ( problemGrade * crimp.holdType ) / crimp.holdSize ) - ropeLength;
+    return difficulty;
+}
+
 /*
  2)
  */
 
 struct TopRopeRoute
 {
     double wallAngle = 10;
-    int moves = 32;
 
     struct RouteGrade
     {
         double gradeNumber = 5.10;
         char gradeLetter = 100; // ascii letter 'd'
     };
 
-    void buildRoute( double wallAngle, RouteGrade grade );
+    void buildRoute( int moves, double wallHeight );
 
     RouteGrade hard;
 };
 
+void TopRopeRoute::buildRoute( int moves, double wallHeight = 40.36 )
+{
+    BoulderProblem::Hold hold;
+
+    for ( int i = moves; i > 0; i-- )
+    {
+       hold.holdHeight = ( wallHeight / moves ) * i;
+    };
```

Comment:
no need for the `;` after the `}`.  it's not a struct. 

- [ ] remove `;` after `}`